security/keycloak-oidc-client-basic/src/test/java/io/quarkus/ts/security/keycloak/oidcclient/basic/BaseOidcClientSecurityIT.java

```diff
@@ -88,6 +88,58 @@ public void adminUserAdminResource() {
                 .body(equalTo("Hello, admin test-admin-user"));
     }
 
+    @Test
+    public void tokenProviderCheckExpirationAndAccess() throws InterruptedException {
+        String token = TokenProviderMethod.TOKEN_PROVIDER.getToken(getApp());
+
+        getApp().given()
+                .when()
+                .auth().preemptive().oauth2(token)
+                .get("/default")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body(equalTo("Hello, user service-account-test-application-client"));
+
+        // Wait until the token expire the 5s expiration time + 5s set via quarkus.oidc.token.lifespan-grace
+        Thread.sleep(10000);
```

Comment:
Would it be better to use something like `Awaitility.await().atMost(10, TimeUnit.SECONDS)...`?

Reply:
Personaly for me using Awaitility is when I want to wait for something which can happen earlier (to safe time) or want to ensure that it happen in some time windows (ensure reliability).

I'm not against it but I think for this usecase when I want to just wait x seconds is `Thread.sleep()` more readable.

If you want me to change it I'll do it.

Reply:
I get your point , and it make sense. 
But, FMHO using would make the test robuster and it can be proceed as soon condition is met, rather waiting the full time, also here in this guide they use it --> https://quarkus.io/guides/security-openid-connect-client-reference#integration-testing-oidc-client... 
For that, I'd suggest switching to Awaitility, but it's up to u, thanks.

Reply:
Updated